docs/guides/rest_authc_authz.md

```diff
@@ -0,0 +1,69 @@
+# rest_authc_authz
+
+## Description
+
+The rest authenticator rely on an external rest endpoint to validate user credentials.
+
+It will:
+
+- check in the system_auth.roles table if the role exist. If the role exist check password with the stored hashed one
+- If the role doesn't exist, it call an external rest endpoint (https only)
+- If creds are fine it stored the user/hashed password in the system_auth.roles table and return the authenticate object user
+
+The rest role manager only query for member roles so the default authorizer can assign permission to the uer role depending to it's member roles
```

Comment:
Can you highlight difference of the rest_role_manager compared to the default one ? (might by worth to add also this information as a comment in the rest_role_manager.hh). Without comparing code of the two implementations it's a bit difficult to find back the rational for having a custom implementation. It will help our successors.